Apps/Examples/Examples/All Examples/SymbolClusteringExample.swift

```diff
@@ -0,0 +1,166 @@
+            100
+            UIColor(red: 0.85, green: 0.11, blue: 0.38, alpha: 1.00)
+        })
+
+        // Add an outline to the icons.
+        clusteredLayer.iconHaloColor = .constant(.init(color: .black))
+        clusteredLayer.iconHaloWidth = .constant(4)
+
+        // Adjust the scale of the icons based on the number of points within an
+        // individual cluster. The first value is a default value.
+        clusteredLayer.iconSize = .expression(Exp(.step) {
+            Exp(.get) { "point_count" }
+            2.5
+            0
+            2.5
+            50
+            3
+            100
+            3.5
+        })
+        print(try! clusteredLayer.jsonObject())
+        return clusteredLayer
+    }
+    
+    func createUnclusteredLayer() -> SymbolLayer {
+        // Create a symbol layer to represent the points that aren't clustered.
+        var unclusteredLayer = SymbolLayer(id: "unclustered-point-layer")
+
+        // Filter out clusters by checking for point_count.
+        unclusteredLayer.filter = Exp(.not) {
+        Exp(.has) { "point_count" }
+        }
+        unclusteredLayer.iconImage = .constant(.name("fire-station-icon"))
+        unclusteredLayer.iconColor = .constant(.init(color: .white))
+
+        // Rotate the icon image based on the recorded water flow.
+        // The `mod` operator allows you to use the remainder after dividing
+        // the specified values.
+        unclusteredLayer.iconRotate = .expression(Exp(.mod) {
+            Exp(.get) { "FLOW" }
+            360
+        })
+        
+        // Double the size of the icon image.
+        unclusteredLayer.iconSize = .constant(2)
+        return unclusteredLayer
+    }
+    
+    @objc func handleTap(gestureRecognizer: UITapGestureRecognizer) {
+        let point = gestureRecognizer.location(in: mapView)
+
+        // Look for features at the tap location within the unclustered layer.
+        mapView.mapboxMap.queryRenderedFeatures(at: point,
+                                                options: RenderedQueryOptions(layerIds: ["unclustered-point-layer", "clustered-fire-hydrant-layer"],
+                                                filter: nil)) { [weak self] result in
+            switch result {
+            case .success(let queriedfeatures):
+                // Return the first feature at that location, then pass attributes to the alert controller.
+                if let firstFeature = queriedfeatures.first?.feature.properties,
+                   let feature = firstFeature["ASSETNUM"] as? String, let location = firstFeature["LOCATIONDETAIL"] as? String {
+                    self?.showAlert(with: "Hydrant \(feature)", and: "\(location)")
+                } else if let firstFeature = queriedfeatures.first?.feature.properties,
+                          let pointCount = firstFeature["point_count"],
+                          let clusterId = firstFeature["cluster_id"] {
```

Comment:
in the previous branch, `ASSETNUM` and `LOCATIONDETAIL` are cast to strings. Should we be doing any casting here? Why is it different?

Also, can add comments to explain what these feature properties represent and where they come from?